api/envoy/config/core/v3/proxy_protocol.proto

```diff
@@ -47,4 +56,40 @@ message ProxyProtocolConfig {
   // This config controls which TLVs can be passed to upstream if it is Proxy Protocol
   // V2 header. If there is no setting for this field, no TLVs will be passed through.
   ProxyProtocolPassThroughTLVs pass_through_tlvs = 2;
+
+  // This config allows additional TLVs to be included in the upstream PROXY protocol
+  // V2 header. Unlike ``pass_through_tlvs``, which passes TLVs from the downstream request,
+  // ``added_tlvs`` provides an extension mechanism for defining new TLVs that are included
+  // with the upstream request. These TLVs may not be present in the downstream request and
+  // can be defined at either the transport socket level or the host level to provide more
+  // granular control over the TLVs that are included in the upstream request.
+  //
+  // Host-level TLVs are specified in the ``metadata.typed_filter_metadata`` field under the
+  // ``envoy.transport_sockets.proxy_protocol`` namespace.
+  //
+  // .. code-block:: yaml
+  //
+  //  clusters:
+  //    - name: cluster_0
+  //      load_assignment:
+  //        endpoints:
+  //          - lbEndpoints:
+  //              - metadata:
+  //                  typed_filter_metadata:
+  //                    envoy.transport_sockets.proxy_protocol:
+  //                      "@type": type.googleapis.com/envoy.config.core.v3.ProxyProtocolConfig
```

Comment:
Thanks for adding this example; this is very useful.

Now that I see this, I notice that there should be a separate config message for this probably, because `ProxyProtocolConfig` contains a bunch of fields that don't apply. It would be a new top-level message, something like

```
message PerHostConfig {
  // Docs: this allows overriding the other thing per-host, etc etc
  repeated TlvEntry added_tlvs = 1;
}
```

@mattklein123 (on behalf of api-shepherds) can you confirm that this is the correct way to do this? There are many examples for `PerRouteConfig` for extensions, but I don't think I've seen `PerHostConfig` before.

Reply:
The initial approach used a separate config message before I moved the API to the ProxyProtocolConfig message to address your previous review comment in https://github.com/envoyproxy/envoy/pull/37591#discussion_r1878682727. Reverting back to that approach & renaming the message works for my use case too. I'll hold off on addressing any of the code-related review comments until we have consensus on the API.

Reply:
Oh, I see. Apologies, I didn't understand then that the message was used as the type for the host metadata, in addition to being used in the transport socket config.

Reply:
This still needs to be fixed

Reply:
Ah, I misunderstood your last comment a couple of weeks back. I'll revert back to the dedicated message approach.

Reply:
The PerHostConfig approach seemed reasonable to me. Implemented that in https://github.com/envoyproxy/envoy/pull/37591/commits/1677c7e9041c339752e6110511f6be8e89ec1a57.

Reply:
@ggreenway One thing to note is that I threw both the config-level `added_tlvs` in the ProxyProtocolConfig message and the PerHostConfig message in the proxy_protocol.proto instead of the upstream_proxy_protocol.proto. My rationale is having both the pass_through_tlvs and added_tlvs fields within the same config made sense to me.